### PR TITLE
Use Neomodel instead of py2neo. [FEEDBACK WANTED]

### PR DESCRIPTION
**Pull request by Askir**
This is how Neomodel would work. 

But in general I think the library is quite a bit better. It also has the ability to have Models for Relationships so we are able to save properties in the "TAGGED_WITH" Relationship for example. Which is really nice imo and was definitely missing in py2neo. 

The interface isn't perfect. For some reason the interal neo id is exposed as .id which is really annoying because now we have to use id_ again to save our own uuid. But at least they provide an easy property for that yay!

Some other things that made me stumble:
- The uuid is saved as hex (without the hypens "-") but when exported via Marshmallow the hypens are added again... So the returned id cannot be searched for in the database. you have to remove the hyphens first.
- All `find` or `all` methods we used before are via the interface `object.nodes.xyz` which again doesn't hide the graph very well. But I think we can provide our own wrappers for that again if we want to.
- In general the interace is more adapted to django than rails so I am not quite as familar with it but I guess @arne-z is going to like it.

### CODE REVIEW

**Comment by Askir** [comment]
Pipeline broke I know it's just a spike PR so everyone can see how it would look like. I didn't delete all references to py2neo yet sorry.

**Comment by Askir** [comment]
@erksch would there be any problems in the frontend if from now on we would ommit the hyphens in uuids?

**Comment by Askir** [comment]
E2E is fixed with this PR in Jona: https://github.com/hpi-sam/ask-your-repository-web/pull/42 just waiting for eriks approval.

I know this is a huge PR again but I'd really appreciate reviews at least on a basic level. To make sure everybody understands what this was about and also to make sure I didn't miss any obvious problems

**Comment by erksch** [comment]
@Askir There would be now problem whatsoever in the frontend. If the ID is consistent in all resources everything will work

File: .circleci/config.yml
Comment (erksch) [inline .circleci/config.yml:7]: What is that?

File: .env.testing
Comment (erksch) [inline .env.testing]: It gets even worse :D whats that?

File: application/__init__.py
Comment (erksch) [inline application/__init__.py]: Why not encrypted?

File: application/controllers/artifacts_controller.py
Comment (erksch) [inline application/controllers/artifacts_controller.py:29]: Why is it called ArtifactBuilder now?

File: application/controllers/artifacts_controller.py
Comment (erksch) [inline application/controllers/artifacts_controller.py:82]: I get that you want the ArtifactBuilder to be a builder pattern for the elastic and neo models, but having this and the rails like Model methods (like find_by etc.) on that Builder as well seems a bit unintuitive. 
So maybe don't call it builder or extract the builder part from the model wrapper part.

File: application/models/artifact.py
Comment (erksch) [inline application/models/artifact.py:10]: Shouldn't this be called NeoArtifact?

**Review by erksch — CHANGES_REQUESTED** [review]

**Comment by erksch** [comment]
@Askir Great work overall! Well tested imo.

File: .circleci/config.yml
Comment (Askir) [reply .circleci/config.yml:7]: Instead of URLs Neomodel uses these Database access links, which are similar to rails database links.
You can check them out here: https://apidock.com/rails/ActiveRecord/Base/establish_connection/class
Here is an example for rails from that site: "postgres://myuser:mypass@localhost/somedatabase".
Explanation is protocol://username:password@url:port
But since we have authentication disabled in CI and development I just left username and password empty.

File: .env.testing
Comment (Askir) [reply .env.testing]: Good catch. I didn't know if leaving empty worked so I added random shit here. gonna remove that :D

File: application/__init__.py
Comment (Askir) [reply application/__init__.py]: Is a pain to setup for development and our database is not available remotely anyways.

File: application/controllers/artifacts_controller.py
Comment (Askir) [reply application/controllers/artifacts_controller.py:29]: The endstate I want that class to be is an ArtifactBuilder. It should encapsulate how the relations between artifacts, tags and teams are connected. Currently it also does the job of syncing with elastic search which is something I'll be working on next. So the class only has one responsibility left.

File: application/controllers/artifacts_controller.py
Comment (Askir) [reply application/controllers/artifacts_controller.py:82]: Well, as I said the class is in a very rough state since it has at least two responsibilities right now. I think it will get much more clear once I extract all the ES stuff from it. Then all the find_by calls for example can also be directly executed on the Artifact and not the ArtifactBuilder. You are right though that the name is not very expressive right now. It's difficult to find one that explains all the responsibilites right now though...

File: application/models/artifact.py
Comment (Askir) [reply application/models/artifact.py:10]: I decided to just call all the models without Neo now. Since those are our default models. I also think Elasticsearch doesn't really need models in the future so having these called Neomodels is just going to lead to confusion with the library name :D

File: application/__init__.py
Comment (Askir) [reply application/__init__.py]: Actually nevermind, the problem seems to have been something else. I enabled it again :)

File: application/controllers/artifacts_controller.py
Comment (Askir) [reply application/controllers/artifacts_controller.py:82]: My overall idea is to:
- Have Artifacts be accessed directly for index/show routes
- Have the artifact builder be a small helper class for creating Artifacts (and maybe updating)
- Have a new class ElasticSearchSynchronizer, that manages populating elasticsearch data. And is called in a post_save hook on the Neo4J models.

I want to do all three in a seperate PR since this one is already a pain to read and very large :D

File: .circleci/config.yml
Comment (erksch) [reply .circleci/config.yml:7]: Ah ok I get it.

**Review by erksch — APPROVED** [review]

**Pull request merged**
